VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitWebviewFragment.java

```diff
@@ -71,12 +77,38 @@ public void onActivityCreated(Bundle savedInstanceState) {
             @Override
             public void onReceivedError(WebView view, int errorCode,
                                         String description, String failingUrl) {
-                int error = errorCode;//for testing only
+                if ( callback != null )
+                    callback.finishProcess();
+                if ( errorCode == HttpStatus.SC_FORBIDDEN || errorCode == HttpStatus.SC_UNAUTHORIZED ){
```

Comment:
I wonder if we should detect 404s here too? Since we do that sometimes to hide the existence of things if you don't have access. cc @nasthagiri